core/math/delaunay_2d.h

```diff
@@ -48,50 +49,50 @@ class Delaunay2D {
 	};
 
 	struct Edge {
-		int edge[2];
+		int points[2];
 		bool bad = false;
 		Edge() {}
 		Edge(int p_a, int p_b) {
-			edge[0] = p_a;
-			edge[1] = p_b;
+			// Store indices in a sorted manner to avoid having to check both orientations later.
+			if (p_a > p_b) {
+				points[0] = p_b;
+				points[1] = p_a;
+			} else {
+				points[0] = p_a;
+				points[1] = p_b;
+			}
 		}
 	};
 
-	static bool circum_circle_contains(const Vector<Vector2> &p_vertices, const Triangle &p_triangle, int p_vertex) {
-		Vector2 p1 = p_vertices[p_triangle.points[0]];
-		Vector2 p2 = p_vertices[p_triangle.points[1]];
-		Vector2 p3 = p_vertices[p_triangle.points[2]];
+	static void get_circum_circle(const Vector<Vector2> &p_vertices, const Triangle &p_triangle, Vector2 &r_circum_center, real_t &r_radius_squared) {
+		Vector2 a = p_vertices[p_triangle.points[1]] - p_vertices[p_triangle.points[0]];
+		Vector2 b = p_vertices[p_triangle.points[2]] - p_vertices[p_triangle.points[0]];
 
-		real_t ab = p1.x * p1.x + p1.y * p1.y;
-		real_t cd = p2.x * p2.x + p2.y * p2.y;
-		real_t ef = p3.x * p3.x + p3.y * p3.y;
+		Vector2 O = (b * a.length_squared() - a * b.length_squared()) / (a.cross(b) * 2.0f);
 
-		Vector2 circum(
-				(ab * (p3.y - p2.y) + cd * (p1.y - p3.y) + ef * (p2.y - p1.y)) / (p1.x * (p3.y - p2.y) + p2.x * (p1.y - p3.y) + p3.x * (p2.y - p1.y)),
-				(ab * (p3.x - p2.x) + cd * (p1.x - p3.x) + ef * (p2.x - p1.x)) / (p1.y * (p3.x - p2.x) + p2.y * (p1.x - p3.x) + p3.y * (p2.x - p1.x)));
+		// Rotating by 90°.
+		real_t temp = O.y;
+		O.y = -O.x;
+		O.x = temp;
 
-		circum *= 0.5;
-		float r = p1.distance_squared_to(circum);
-		float d = p_vertices[p_vertex].distance_squared_to(circum);
-		return d <= r;
+		r_radius_squared = O.length_squared();
+		r_circum_center = O + p_vertices[p_triangle.points[0]];
 	}
 
-	static bool edge_compare(const Vector<Vector2> &p_vertices, const Edge &p_a, const Edge &p_b) {
-		if (p_vertices[p_a.edge[0]].is_equal_approx(p_vertices[p_b.edge[0]]) && p_vertices[p_a.edge[1]].is_equal_approx(p_vertices[p_b.edge[1]])) {
-			return true;
-		}
+	static Triangle create_triangle(const Vector<Vector2> &p_vertices, const int &p_a, const int &p_b, const int &p_c) {
+		Triangle result(p_a, p_b, p_c);
 
-		if (p_vertices[p_a.edge[0]].is_equal_approx(p_vertices[p_b.edge[1]]) && p_vertices[p_a.edge[1]].is_equal_approx(p_vertices[p_b.edge[0]])) {
-			return true;
-		}
+		// Get the values of the circumcircle and store them inside the triangle object.
+		get_circum_circle(p_vertices, result, result.circum_centre, result.circum_radius_squared);
 
-		return false;
+		return result;
 	}
 
 	static Vector<Triangle> triangulate(const Vector<Vector2> &p_points) {
 		Vector<Vector2> points = p_points;
 		Vector<Triangle> triangles;
 
+		// Get the bounding rect.
```

Comment:
We can check if the point count is 0-2 initially and return the empty triangles vector right away. Then we'll know there's a "first" point and we can initialize the rect right away, no need for the `if (i == 0)` condition.

But it probably won't amount to anything substantial other than in the 0-2 points case.